de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/views/buttons/ConnectButton.java

```diff
@@ -123,54 +142,202 @@ private void createAddAccountMenuItem() {
+            }
+
+
+        } catch (IllegalAWTContextException e) {
+            LOG.error("Account creation failed.", e);
+
+            SafeDialogUtils.showError(
+                "There was an error creating the account.\nDetails: "
+                    + e.getMessage(),
+                "Account creation failed");
+        }
+    }
+
     /**
-     * Asks for Name, Password and domain and server for a new XMPP account.
+     * Asks for user id (name@domain), password, and server for a new XMPP
+     * account.
+     *
+     * @return an <code>XMPPAccount</code> with the values entered by the user
+     *         or <code>null</code> if the creation was canceled by the user or
+     *         the user entered at least one illegal value
      */
-    protected XMPPAccount createNewAccount() {
-        final String userID = SafeDialogUtils.showInputDialog(
-            "Your User-ID, e.g. user@saros-con.imp.fu-berlin.de", "", "Login");
-        if (userID.isEmpty()) {
+    private XMPPAccount createNewAccount() {
+        final String userID;
+
+        try {
+            userID = SafeDialogUtils.showInputDialog(
+                "Your User-ID, e.g. user@saros-con.imp.fu-berlin.de",
+                "", "Login");
+
+        } catch (IllegalAWTContextException e) {
+            LOG.error("Account creation failed.", e);
+
+            SafeDialogUtils.showError(
+                "There was an error creating the account.\nDetails: "
+                    + e.getMessage(), "Account creation failed");
+
             return null;
         }
-        if (!userID.contains("@")) {
-            SafeDialogUtils.showError("No @ found in the ID!", "Error");
+
+        if (userID == null) {
+            LOG.debug("Account creation canceled by user during user id"
+                + " entry.");
+
             return null;
         }
 
-        String[] fields = userID.split(USERID_SEPARATOR);
-        String username = fields[0];
-        String domain = fields[1];
+        JID jid = new JID(userID);
+
+        if (!jid.isValid() || jid.getName().isEmpty()) {
+            SafeDialogUtils.showError("Entered user id is not valid.",
+                "Account creation aborted");
+
+            LOG.debug("Account creation failed as the user did not provide a "
+                + "valid user id.");
 
-        final String password = Messages
-            .showPasswordDialog("Password", "Password");
-        if (password.isEmpty()) {
             return null;
         }
 
-        // TODO query port
-        String server = SafeDialogUtils.showInputDialog(
-            "XMPP server (optional, not necessary in most cases)", "",
-            "Server");
+        String username = jid.getName();
+        String domain = jid.getDomain();
+
+        final String password;
 
         try {
-            return accountStore
-                .createAccount(username, password, domain, server, 0, true,
-                    true);
+            password = SafeDialogUtils
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: 'Password' detected in this expression, review this potentially hard-coded credential. ([squid:S2068](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS2068))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=240&issues=AWYbBs1KvOWp7Yd63TpK&open=AWYbBs1KvOWp7Yd63TpK)

[ISSUE_KEY]: AWYbBs1KvOWp7Yd63TpK